### ISSUE TITLE
PyPlot works by calling matplotlib  directly crashes julia silently

### ISSUE THREAD

**Issue by musm** [JuliaPy/PyCall.jl#665]
```julia

julia> using PyCall

julia> plt = pyimport("matplotlib.pyplot")
PyObject <module 'matplotlib.pyplot' from 'C:\\tools\\Miniconda3\\lib\\site-packages\\matplotlib\\pyplot.py'>

julia> x = range(0; stop=2*pi, length=1000); y = sin.(3 * x + 4 * cos.(2 * x));

julia> plt.plot(x, y)
This application failed to start because it could not find or load the Qt platform plugin "windows"
in "".

Reinstalling the application may fix this problem.

```
This causes Julia to terminate abruptly. 

Yet using PyPlot works ?

**Comment by stevengj** [comment]
PyPlot works because it uses a different backend, presumably.

Some of the Matplotlib backends only seem to work with the `python` executable, though I'm not sure why in this particular case — PyPlot usually works with Qt.

**Comment by musm** [comment]
@stevengj  I did some investigating to see if  loading the matplotlib package with PyCall vs loading it with PyPlot  was somehow loading the wrong dll loading to a crash, using process explorer. 

However that confirmed BOTH methods loaded the **exact same  dll's**  (backend for both is QT5)

**Comment by tkf** [comment]
Maybe call `pygui_start(:qt_pyqt5)`? That's another difference to PyPlot. Although I guess it only matters if you have non-default Qt configuration.

**Comment by stevengj** [comment]
By default, Matplotlib starts in non-interactive mode, so you shouldn't need a `pygui` event loop in Julia.  The plot window doesn't appear until you run `plt.show()`, in which case Python implements the event loop.

**Comment by stevengj** [comment]
There are multiple Python Qt libraries.  Maybe PyPlot is importing a different one and Matplotlib is respecting that?  See https://github.com/JuliaPy/PyPlot.jl/blob/master/src/init.jl

**Comment by musm** [comment]
It's quite strange because calling matplotlib from python works, PyPlot works, just from PyCall it fails.

**Comment by tkf** [comment]
`pygui_start(:qt_pyqt5)` calls `fixqtpath` before starting the event loop. I thought it could be doing something.

**Comment by stevengj** [comment]
Oh right, I forgot about `fixqtpath()`.  You could definitely try calling `PyCall.fixqtpath()` manually.

**Comment by musm** [comment]
Yes calling `PyCall.fixqtpath()` manually after  import matplotlib fixes the issue.  @tkf it seems `PyCall.fixqtpath()` is good enough to avoid the crash. 

We'll the error ends up  killing julia, which is pretty annoying. Is that avoidable?

**Comment by tkf** [comment]
Yeah, it'd be nice if we can avoid it.  Though I don't know what's the best way to do it.  Is it too crazy to always call `fixqtpath` in `PyCall.__init__`?

**Comment by musm** [comment]
Doesn't seem too crazy, until a proper fix is implemented.  Are there any other instances that can crash julia like this  or is it only the QT path loading issue.

**Comment by stevengj** [comment]
I guess since `fixqtpath` just sets an environment variable, it is not so bad to run unconditionally on startup.  (Only mildly insane, but we are used to that. 🙄 )

Any library that calls `os.abort()` or similar will terminate Julia (or any other program).  This is not a crash, and it is impossible to prevent in general.

**Comment by tkf** [comment]
> Only mildly insane, but we are used to that.

:rofl: 

> The Qt folks seem to think that the location of `qt.conf` is not their problem. https://bugreports.qt.io/browse/QTBUG-58637

Reporting it to all four Python Qt bindings sounds like a nightmare...  Especially if just a subset of them support it.

I guess the upside is that Qt team seems at least positive for solving the "root cause" https://bugreports.qt.io/browse/QTBUG-14150

**Comment by musm** [comment]
Seems like this is an issue they will never fix 😄
